src/main/java/com/internship/backend/controller/UserController.java

```diff
@@ -24,13 +25,20 @@
 
 @RestController
 @RequestMapping("api/users")
+@RequiredArgsConstructor
```

Comment:
Could you explain what this annotation does? What itis used for?

Reply:
I used this annotation in UserController class to handle dependency injection. Lombok will generate a constructor that takes UserService as an argument.
Instead of 
**@Autowired
private UserService userService**   (in this case, @Autowired handles the dependency injection)

I should have written
**private final UserService userService**
to be able to use the @RequiredArgsConstructor